### PR TITLE
Warn about invalid arguments to ColorFilter.mode

### PR DESCRIPTION
**Pull request by jason-simmons**
Fixes https://github.com/flutter/flutter/issues/100702

### CODE REVIEW

File: lib/web_ui/lib/src/engine/canvaskit/color_filter.dart [hunk 1]
```diff
@@ -83,10 +83,14 @@ class CkBlendModeColorFilter extends CkColorFilter {
 
   @override
   SkColorFilter _initRawColorFilter() {
-    return canvasKit.ColorFilter.MakeBlend(
+    SkColorFilter filter = canvasKit.ColorFilter.MakeBlend(
```
Comment (yjbanov) [inline lib/web_ui/lib/src/engine/canvaskit/color_filter.dart:86]: If `MakeBlend` is allowed to return null, then we should change the API binding in `canvaskit_api.dart` to return `SkColorFilter?`.

File: lib/web_ui/lib/src/engine/canvaskit/color_filter.dart [hunk 2]
```diff
@@ -83,10 +83,14 @@ class CkBlendModeColorFilter extends CkColorFilter {
...
       toSharedSkColor1(color),
       toSkBlendMode(blendMode),
     );
+    if (filter == null) {
+      throw ArgumentError('Invalid parameters for blend mode ColorFilter');
```
Comment (yjbanov) [inline lib/web_ui/lib/src/engine/canvaskit/color_filter.dart:91]: Do we know what combinations of color/blendMode we support? The message would be more helpful if we could give the user more information.

File: lib/web_ui/lib/src/engine/canvaskit/color_filter.dart (hunk 1 above)
Comment (jason-simmons) [reply lib/web_ui/lib/src/engine/canvaskit/color_filter.dart:86]: Done

File: lib/web_ui/lib/src/engine/canvaskit/color_filter.dart (hunk 2 above)
Comment (jason-simmons) [reply lib/web_ui/lib/src/engine/canvaskit/color_filter.dart:91]: The rules for which combinations are usable are implemented in https://github.com/google/skia/blob/main/src/core/SkModeColorFilter.cpp#L126

AFAIK Skia does not provide a concise summary of the valid arguments for this function.

**Review by yjbanov — APPROVED** [review]

**Pull request merged**